### PR TITLE
8289051: C2: Cleanup PhaseCCP::analyze()

### PR DESCRIPTION
**Pull request by chhagedorn**
As part of [JDK-8288683](https://bugs.openjdk.org/browse/JDK-8288683), which went into JDK 19, I separated out most of the included refactorings into this cleanup RFE to make the fix as small as possible.

This cleanup includes:
- More comments and better descriptions.
- Renamed some variables.
- Split code into multiple methods to improve the readability.

Thanks,
Christian

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289051](https://bugs.openjdk.org/browse/JDK-8289051): C2: Cleanup PhaseCCP::analyze()

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [5ba8f725](https://git.openjdk.org/jdk/pull/9833/files/5ba8f72586254727b6cf97c4438059a09a254176)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9833/head:pull/9833` \
`$ git checkout pull/9833`

Update a local copy of the PR: \
`$ git checkout pull/9833` \
`$ git pull https://git.openjdk.org/jdk pull/9833/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9833`

View PR using the GUI difftool: \
`$ git pr show -t 9833`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9833.diff">https://git.openjdk.org/jdk/pull/9833.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/phaseX.cpp [hunk 1]
```diff
@@ -1753,11 +1753,15 @@ static bool ccp_type_widens(const Type* t, const Type* t0) {
 }
 #endif //ASSERT
 
-//------------------------------analyze----------------------------------------
+// In this analysis, all types are initially set to TOP. We iteratively call Value() on all nodes of the graph until
+// we reach a fix-point (i.e. no types change anymore). We start with a list that only contains the root node. Each time
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/phaseX.cpp:1757]:
```suggestion
// we reach a fixed-point (i.e. no types change anymore). We start with a list that only contains the root node. Each time
```

**Review by TobiHartmann — APPROVED** [review]
Nice cleanup. Looks good to me.

**Comment by chhagedorn** [comment]
Thanks Tobias for your review!

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by chhagedorn** [comment]
Thanks Vladimir for your review!

/integrate

**Pull request closed without merging**